Formula/eiffelstudio.rb

```diff
@@ -32,9 +33,15 @@ def install
 
     os = OS.mac? ? "macosx" : OS.kernel_name.downcase
     os_tag = "#{os}-x86-64"
-    system "./compile_exes", os_tag
+    begin
+      system "./compile_exes", os_tag
+    rescue
+      # TODO: debugging Linux failure...
+      system "cat", "compile.log"
+      raise
+    end
```

Comment:
After printing the compilation log file, the Linux failure is:
```
ld -m elf_x86_64 -shared -o libmtfinalized.so ../idrs/MTidrs.o MTscoop.o MTscoop_gc.o MTscoop_helpers.o MTprocessor_registry.o MTprivate_queue.o MTprocessor.o MTqueue_cache.o MTmessage_channel.o MTrequest_group.o MTidentifier_set.o MTmalloc.o MTgarcol.o MTlocal.o MTstack.o MTstore.o MTretrieve.o MThash.o MTtraverse.o MThashin.o MTtools.o MTinternal.o MTplug.o MTcopy.o MTequal.o MTlmalloc.o MTout.o MTtimer.o MTurgent.o MTsig.o MThector.o MTcecil.o MTfile.o MTdir.o MTmisc.o MTstring.o MTerror.o MTumain.o MTmemory.o MTmemory_analyzer.o MTargv.o MTboolstr.o MTsearch.o MTrun_idr.o MTpath_name.o MTobject_id.o MTeif_threads.o MTeif_project.o MTposix_threads.o MTgen_conf.o MTeif_type_id.o MTrout_obj.o MToption.o MTcompress.o MTconsole.o MToffset.o MTmain.o MTexcept.o  -lm -lpthread
  ld: cannot find -lm: No such file or directory
  ld: cannot find -lpthread: No such file or directory
```

Reply:
It might make sense for `-lpthread` not to work as I think Ubuntu 22.04 is trying to move towards to `-pthread` instead.  But `-lm` is not to my knowledge unsupported.  I am taking a look at this.

Reply:
Oddly, the old version builds fine, so the failure here is specific to the newer version.

Reply:
I pushed a fix that worked for me locally.  For some reason directly invoking `ld` causes the linking step to fail, but using `ENV.cc` (and removing the unneeded `-m` flag) works with no issues.  I'm not really sure why they are invoking `ld` directory like this as it is probably more portable to let GCC figure out how to call the linker instead of invoking it directly.

Reply:
Can we report this upstream?

Reply:
Yes, I will open a pull request and link it in the comment.  One thing that is unclear at the moment is that it may be that the config file we are modifying is intended to be customized by the user, so it may be more of a question of what is the best default.  I'm also going to check what changed from the previous release we were using because as mentioned above that builds with no issues.

Reply:
I opened bug report and added a link in a comment, as the GitHub only appears to be a mirror.

Reply:
Looks like debugging commit got re-added on rebase. Need to drop that before merge.